### ISSUE TITLE
Undefined method '[]' for nil:NilClass

### ISSUE THREAD

**Issue by michaelsking1993** [guilhermesad/rspotify#138]
[disclaimer: apologies if this is already covered, I didn't see it - will remove immediately if so]

Trying to connect with the gem, I tried out the first example from the doc:

```
require 'rspotify'
artists = RSpotify::Artist.search('Arctic Monkeys')

```
And I get the following error:
```
NoMethodError: undefined method '[]' for nil:NilClass
from /Users/michael/.rvm/gems/ruby-2.4.1/gems/rspotify-1.24.0/lib/rspotify/base.rb:109:in `block in search'
```

Would love to help out with this if I can.

**Comment by estiens** [comment]
This usually means you have to authenticate first. I agree better error handling on the response from Spotify rather than going straight to a nil error would be better here though.

**Comment by guilhermesad** [comment]
Solved in #170.
